### ISSUE TITLE
Default args should not be treated as magic numbers

### ISSUE THREAD

**Issue by marines** [palantir/tslint#1965]
### Bug Report

- __TSLint version__: `4.2.0`
- __TypeScript version__: `2.1.4`
- __Running TSLint via__: gulp-tslint

#### TypeScript code being linted

```ts
getPage(page: number = 1, perPage: number = 200): angular.IPromise<string> {
```

with `tslint.json` configuration:

```json
"no-magic-numbers": true
```

#### Actual behavior

'magic numbers' are not allowed (no-magic-numbers)

Note that the error is right before the `200` number.

#### Expected behavior

This should be valid as the argument name names the number so it's not magic anymore.
